### DISCUSSION TITLE
Does translating technical docs cause more harm than good?

### DISCUSSION THREAD

**Discussion by tamaracha** [withastro/docs#6554] [Ideas]
Actually, this is a more general and even cultural question, but I want to address it here, because Astro has a motivated community and very beautiful docs. ;-)

As a German with much experience in software development and its workflows, I don't get any benefit from german translation of technical docs at all. I think that translating introductory content, tutorials, pros and cons etc. is a good idea to become more approachable for newcomers. But are there any advantages in having docs in different languages? I don't know any germans who really enjoy reading translated docs, but I'm interested in opinions.

1. The process of reading docs is highly top-down. You know what you're looking for, you already have some concepts in mind and try to match the docs with these concepts quickly. Having multiple terms for the same concept requires more cognitive effort and makes this process inefficient.
2. Search engine results get polluted with duplicated entries, e.g., styles and styling. Microsoft made it impossible to find relevant docs for their older products, because they have a huge translation mess.
3. If something doesn't work as expected, you need to search for solutions and suggestions. Looking for the original term reveals way more resources like discussions, questions, solutions, blog articles. If we get used to the translated terms, we have to look up the original term before searching for resources. The tech stuff works the same everywhere, segregation by language doesn't make sense.
4. Maybe this is a cultural thing, but the tonality doesn't feel natural. I can easily accommodate to the more enthusiastic style in written english, it's a social context switch. But transplanting this style to german feels weird. Many sites have started to offer translated content, probably because AI driven translation has become so easy. Most of this content feels like a phone call in a violin concert, it's painful. In the Astro docs, I can see that the translators put much effort in good style (thank you), but this subtle discomfort with written tech german remains. Maybe it's even Microsoft who built the ground for this discomfort with their terrible style that sounds like a bureaucracy parody.

To my mind, it makes more sense to leave the docs in english and pay more attention on translating getting-started tutorials, recipes, audience-targeted articles etc. When I worked with university staff, I realized that many scientists don't know what technical solutions can do for them, what are their limits, why we should consider using them in our specific situation. Their English is good enough for writing and reading papers, so reading how something works shouldn't be such a big deal. Their problem is, e.g., they spend much time with manually testing their matlab functions and are scared of messing-up their code. I had to introduce the concept of unit testing which brings more confidence and a quicker workflow. I had to introduce git because you can break your code and go back to a working version. Astro and other SSG make it easy to create small and fast static sites for projects, events, research groups etc. without relying on Google Sites. After establishing the basic concepts with localized content, reading english docs the top-down way is no problem.

Thanks for reading, and please share your opinion.

**Comment by tamaracha** [comment]
There's one more serious issue with translated tech content: Criminal Clickbait Sites. If people search for tech questions or problems in their language, they get search results from sites that steal content from official resources and offer translated versions, just for getting clicks, showing ads, spreading misinformation, installing malware etc. The domains look similar to official sites, e.g., `stackovercoder.com.de`. With translation AI becoming better, recognizing auto-translated content became more difficult, so readers are in higher risk to open such clickbait results. Sometimes, it's just a strange metaphore that makes the difference, whereas the grammar can be correct. In order to avoid this SEO BS, you have to search in English or read trusted sources in your language like magazines, forums etc.

**Comment by afucher** [comment]
Here in Brazil, a lot of people that works with technology or wants to work with, do not know english or just a little bit. So, to introduce people and make the technology more inclusive, having translated docs is very useful.

**Comment by tamaracha** [comment]
Thanks @afucher for your reply. Still this sounds like having translated introductory material/tutorials could bridge that gap very well. Those are important for being approachable and building basic concepts and understanding. When the basics are established, reading docs becomes scanning from term to term, and the English between the terms is noise that your brain learns to ignore unconsciously. Actually, native tech-educational material citing references in  english offers a valuable opportunity to obtain some basic English skills. Children can inductively infer the meaning of terms by matching e.g. code snippets and instructions from their native resources with the English docs. This inductive learning is much more fun than learning by heart from a text book at school, and I know many people who learned their first rudimentary English this way via coding. Even basic skills with bad grammar are useful in many situations. If you keep English away from people, it remains scary for them. From a long-term perspective, hiding English is not inclusive but locks people in their area.
